src/animals/Animal.java

```diff
@@ -0,0 +1,127 @@
+
+    public static Island getIsland() {
+        return island;
+    }
+
+    public static void setIsland(Island island) {
+        Animal.island = island;
+    }
+
+
+    abstract public void dieFromStarvation();
+
+    public int getMaxStepsPerMove() {
+        return maxStepsPerMove;
+    }
+
+    public void setMaxStepsPerMove(int maxStepsPerMove) {
+        this.maxStepsPerMove = maxStepsPerMove;
+    }
+
+    public int getY() {
+        return y;
+    }
+
+    public void setY(int y) {
+        this.y = y;
+    }
+
+    public int getX() {
+        return x;
+    }
+
+    public void setX(int x) {
+        this.x = x;
+    }
+
+    public String getPictureOfAnimal() {
+        return pictureOfAnimal;
+    }
+
+    public void setPictureOfAnimal(String pictureOfAnimal) {
+        this.pictureOfAnimal = pictureOfAnimal;
+    }
+
+    public String getNameOfAnimal() {
+        return nameOfAnimal;
+    }
+
+    public void setNameOfAnimal(String nameOfAnimal) {
+        this.nameOfAnimal = nameOfAnimal;
+    }
+
+    public double getWeightOfAnimal() {
+        return weightOfAnimal;
+    }
+
+    public void setWeightOfAnimal(double weightOfAnimal) {
+        this.weightOfAnimal = weightOfAnimal;
+    }
+
+    public double getKilogramsOfFoodForCompleteSaturation() {
+        return kilogramsOfFoodForCompleteSaturation;
+    }
+
+    public void setKilogramsOfFoodForCompleteSaturation(double kilogramsOfFoodForCompleteSaturation) {
+        this.kilogramsOfFoodForCompleteSaturation = kilogramsOfFoodForCompleteSaturation;
+    }
+
+    public double getKilogramsOfFoodsInTheStomach() {
+        return kilogramsOfFoodsInTheStomach;
+    }
+
+    public void setKilogramsOfFoodsInTheStomach(double kilogramsOfFoodsInTheStomach) {
+        this.kilogramsOfFoodsInTheStomach = kilogramsOfFoodsInTheStomach;
+    }
+
+    public boolean isAlive() {
+        return isAlive;
+    }
+
+    public void setAlive(boolean alive) {
+        isAlive = alive;
+    }
+
+    public abstract boolean eat(Animal animal);
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Animal animal = (Animal) o;
+        return timeDelay == animal.timeDelay && Double.compare(animal.weightOfAnimal, weightOfAnimal) == 0 && Double.compare(animal.kilogramsOfFoodForCompleteSaturation, kilogramsOfFoodForCompleteSaturation) == 0 && Double.compare(animal.kilogramsOfFoodsInTheStomach, kilogramsOfFoodsInTheStomach) == 0 && isAlive == animal.isAlive && maxStepsPerMove == animal.maxStepsPerMove && y == animal.y && x == animal.x && Objects.equals(newbornAnimals, animal.newbornAnimals) && Objects.equals(pictureOfAnimal, animal.pictureOfAnimal) && Objects.equals(nameOfAnimal, animal.nameOfAnimal);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(pictureOfAnimal, nameOfAnimal);
+    }
+
+    public abstract boolean eat(Plant plant);
```

Comment:
move all abstract methods on the top, after that declare all methods with implementations